core/common/src/main/scala/org/typelevel/otel4s/TextMapUpdater.scala

```diff
@@ -45,3 +47,23 @@ trait TextMapUpdater[A] {
     */
   def updated(carrier: A, key: String, value: String): A
 }
+
+object TextMapUpdater {
+  def apply[A](implicit updater: TextMapUpdater[A]): TextMapUpdater[A] = updater
+
+  implicit def forMap[CC[x, +y] <: MapOps[x, y, CC, CC[x, y]]]
```

Comment:
looking at this some more, I think it wouldn't work for an implementation that is specialised for `String` keys. not entirely sure how to tweak it to support that

Reply:
after thinking about this, I realised that the type signatures don't allow/support such an implementation regardless, so it's not an issue